### ISSUE TITLE
Show vehicle part location (and search?)

### ISSUE THREAD

**Issue by narc0tiq** [nornagon/cdda-guide#133]
Vehicle parts have an install "location" within their vehicle tile (e.g., `"location": "center"` in the raw JSON on [vehicle_part/animal_compartment](https://cdda-guide.nornagon.net/vehicle_part/animal_compartment)). These define exclusion groups: no two vehicle parts with the same `location` can be installed in the same tile.

As far as I know, the locations themselves are completely freeform strings, but it would still be nice to be able to see the opportunity cost of installing, say, a 60L tank (in location `fuel_source`: it blocks, among other things, a large storage battery from being in the same tile).

**Comment by nornagon** [comment]
showing the location is pretty straightforward, so i did that. listing out parts by their location is a little more difficult, as "vehicle_part_location" isn't really a "kind of thing" in the JSON; it's just a string. I could _pretend_ that it was a kind of thing and make /vehicle_part_location/center show a list of all vehicle parts with the location "center", but I'm not sure I really want to do that. The [main vehicle part list](https://cdda-guide.nornagon.net/vehicle_part) categorizes by "categories", and there's no obvious way to show multiple axes here (category then location? location then category? a toggle? all could work, none is clearly best).

Perhaps someday vehicle part locations will become real things in the JSON (e.g. it would be useful to store their Z-order in! right now that's a [hard-coded table](https://github.com/CleverRaven/Cataclysm-DDA/blob/0b9b1fa700787f65a85773824dffaad161503849/src/veh_type.cpp#L756-L798). For now, well, at least you can see the location on the part page? :)

**Comment by narc0tiq** [comment]
> it's just a string

Yep, you'd have to build up your own index in some kind of preprocessing step. I'm not sure it's worth doing, but maybe something to keep in mind in some future when you have a bunch of time?

> no obvious way to show multiple axes

Honestly, the easy answer is not to do it, of course! Otherwise, a dropdown or set of radio buttons could allow the user to choose between different view options, but that would make it a very unusual listing from all the others. Eh... who knows, maybe you could find a use for this dropdown capability in the other listings? Sort item types by, uh... volume? Monsters by evolution depth*? Dunno, I feel there's potentially something there but also clearly not something _necessary_, if you know what I mean.

> at least you can see the location on the part page

Yes! That honestly goes a long way towards being able to decide what should go where, but also to diagnose why something can't go somewhere. As an example of the latter, I was very shocked (pun not intended) to find that _shock absorbers_ are mutually exclusive with _armor plating_.

---

* I'm not very good at graphviz, but I wonder how horrible it would be to make a diagram with the graph of zombie evolutions. I have a suspicion it would look really cool, but probably not nicely broken down into "tiers", if you know what I mean.
